gitbook/gitbookCertification.go

```diff
@@ -13,10 +13,31 @@ func (openControl *OpenControlGitBook) exportControl(control *ControlGitbook) (s
 	key := replaceParentheses(fmt.Sprintf("%s-%s", control.standardKey, control.controlKey))
 	text := fmt.Sprintf("#%s\n##%s\n", key, control.Name)
 	openControl.Justifications.GetAndApply(control.standardKey, control.controlKey, func(selectJustifications models.Verifications) {
+		// In the case that no information was found period for the standard and control
+		if len(selectJustifications) == 0 {
+			errorText := fmt.Sprintf("No information found for the combination of standard %s and control %s", control.standardKey, control.controlKey)
+			text = fmt.Sprintf("%s%s\n", text, errorText)
+			return
+		}
 		for _, justification := range selectJustifications {
 			openControl.Components.GetAndApply(justification.ComponentKey, func(component base.Component) {
 				text = fmt.Sprintf("%s\n#### %s\n", text, component.GetName())
-				text = fmt.Sprintf("%s%s\n", text, justification.SatisfiesData.GetNarrative())
+
+				if component.GetResponsibleRole() != "" {
+					text = fmt.Sprintf("%s\n##### Responsible Role: %s\n", text, component.GetResponsibleRole())
+				}
+
+				if len(justification.SatisfiesData.GetNarratives()) == 0 {
+					text = fmt.Sprintf("%sNo narrative found for the combination of standard %s and control %s\n", text, control.standardKey, control.controlKey)
+					return
+				}
+
+				for _, narrative := range justification.SatisfiesData.GetNarratives() {
+					if narrative.GetKey() != "" {
+						text = fmt.Sprintf("%s\n##### %s\n", text, narrative.GetKey())
+					}
+					text = fmt.Sprintf("%s%s\n", text, narrative.GetText())
+				}
```

Comment:
Ditto with the outer callback. Basically, break `exportControl()` into smaller pieces.